test/test_tune.py

```diff
@@ -135,30 +135,54 @@ def test_autotune_match(self, open_db_mock: mock.Mock, close_db_mock: mock.Mock,
         context.device.driver_version = 'another version'
         context.device.platform_name = 'another platform'
         context.device.name = 'another device'
-        ret3 = self.autotune(context, 'xyz')
-        assert ret3 == another_tuning
+        ret = self.autotune(context, 'xyz')
+        assert ret == another_tuning
+
+        # Populate with an empty parameter set
```

Comment:
I don't think this is testing the condition that caused the bug in the first place. The parameter set is `('',)`, which is not the same as `()`.